staging/src/k8s.io/apiserver/pkg/authorization/config/register.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GroupName is the group name use in this package
+const GroupName = "apiserver.config.k8s.io"
```

Comment:
I realized while reviewing (https://github.com/kubernetes/kubernetes/pull/118984#discussion_r1297536742) that we already have a package for apiserver config types that has all the registration/generation boilerplate set up already

relocate the API types to the existing types.go files in API server config packages staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1 and staging/src/k8s.io/apiserver/pkg/apis/apiserver and add a single line registering `&AuthorizationConfiguration{}` in staging/src/k8s.io/apiserver/pkg/apis/apiserver/register.go and staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/register.go